### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by growingananas**
[DEMO LINK](https://growingananas.github.io/react_tabs/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
-export const tabs = [
+export const tabs: Tab[] = [
   { id: 'tab-1', title: 'Tab 1', content: 'Some text 1' },
   { id: 'tab-2', title: 'Tab 2', content: 'Some text 2' },
   { id: 'tab-3', title: 'Tab 3', content: 'Some text 3' },
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTabId] = useState('tab-1');
+  const [selectedTabTitle, setSelectedTabTitle] = useState('Tab 1');
```
Comment (silchyn) [inline src/App.tsx:15]: So we just duplicate the logic everywhere?

Comment (silchyn) [inline src/App.tsx:15]: There are some magic strings

File: src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -1 +1,53 @@
...
+};
+
+export const Tabs: React.FC<Props> = ({
+  selectedTabId,
+  onTabSelected,
+  tabs,
+}) => {
+  const currentTabId = tabs
+    .find(tab => tab.id === selectedTabId)?.id
+    || tabs[0].id;
```
Comment (silchyn) [inline src/components/Tabs/Tabs.tsx:16]:
```suggestion
  const currentTabId = (tabs.find(tab => (
    tab.id === selectedTabId
  )) || tabs[0]).id;
```

File: src/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -1 +1,53 @@
...
+                'is-active': currentTabId === tab.id,
+              })}
+              data-cy="Tab"
+            >
+              <a
+                href={`#${tab.id}`}
+                data-cy="TabLink"
+                onClick={() => {
+                  return currentTabId !== tab.id ? onTabSelected(tab) : null;
+                }}
```
Comment (silchyn) [inline src/components/Tabs/Tabs.tsx:35]: Put it in a separate callback

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/components/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -1 +1,53 @@
...
+export const Tabs: React.FC<Props> = ({
+  tabs,
+  selectedTabId,
+  onTabSelected,
+}) => {
+  const currentTab = tabs.find(tab => (
+    tab.id === selectedTabId
+  )) || tabs[0];
+
+  const handleSelectedTab = (tab: Tab) => {
```
Comment (33Orange) [inline src/components/Tabs/Tabs.tsx:18]:
```suggestion
  const selectTab = (tab: Tab) => {
```

**Review by 33Orange — APPROVED** [review]
Well done

**Pull request closed without merging**